packages/styled/types/tests.tsx

```diff
@@ -24,25 +24,13 @@ styled.svg
 const StyledDiv = styled.div({})
 ;<StyledDiv ref={React.createRef()} />
 
-// can specify theme for StyledTags
-const themedStyled = styled as CreateStyled<{ themeProp: string }>
-const StyledDiv3 = themedStyled.div`
-  color: ${props => props.theme.themeProp}
-`
-;<StyledDiv3 />
-const StyledDiv2 = themedStyled.div(props => {
-  // $ExpectType { themeProp: string; }
-  props.theme
-
-  return {}
-})
-;<StyledDiv2 />
-
 // Can override theme optionally on prop
-;<StyledDiv2 theme={{ themeProp: '' }} />
+// TODO: Fix types for this usage
+// $ExpectError
+;<StyledDiv theme={{ primary: '' }} />
```

Comment:
This here currently errors, because when defining theme this way, one has to implement for the full spec of Theme.

_I'm a little tired to think of solutions from my end, can chime in tomorrow. Just exposing this issue for discussion._

Reply:
Not sure if this is incorrect though. If `StyledDiv` depends on `Theme` we can't know on which parts of the `Theme` it actually depends, so technically putting an incomplete theme here could be a mistake.

Reply:
Yeah, with a clear mind today, I totally agree - didn't really think it through yesterday.